### ISSUE TITLE
Default Types setting for Add Block interface

### ISSUE THREAD

**Issue by joe9663** [BoldGrid/post-and-page-builder#449]
**Is your feature request related to a problem? Please describe.**
If a user wants to reuse their saved blocks frequently they have to change the Types dropdown from All Types to Block Library every time they open the Add Block interface.

**Describe the solution you'd like**
Add a configuration to the Post and Page Builder Settings page to allow users to choose the default type that will be selected when they open the Add Block interface. We could also change the behavior automatically for them, like if there are no Saved Blocks then All Types is the default selection but if any Saved Blocks are detected then Block Library is chosen instead.

![image](https://user-images.githubusercontent.com/17650704/189362561-00967d4a-ced0-414c-a23e-cf571d7a947e.png)

**Comment by joe9663** [comment]
@jamesros161 +1 for this from UserEcho

**Comment by jamesros161** [comment]
@joe9663 what if we set it to open up to the last filter they had selected?

**Comment by joe9663** [comment]
That would work too. I think it should be a global thing instead of a per page one. Like if they selected Block Library on one page and started editing another page then Block Library would be selected automatically.

**Comment by jamesros161** [comment]
I agree.  I think that whenever they select an option in the filter, it will save to the user's meta. Then, when they click the 'Add Block'  button, it'll check that user_meta field
